### PR TITLE
Add CronJob KSM metrics.

### PR DESCRIPTION
**Pull request by htroisi**
### Summary

The Kubernetes team is working on adding new metrics for all Kubernetes workloads, including `CronJob`. This PR:
- Adds the stable KSM CronJob metrics to the metric definition file.
- Updates the e2e integration k8s spec.

### Related PRs
- Update testing resources to include a `CronJob` workload: #608 #622
- Update static test data to include KSM `CronJob` metrics: #623
- Update `infra-integration-specs` [PR](https://source.datanerd.us/infrastructure/infra-integration-specs/pull/302) 
- Update `infra-metrics-utils` [PR](https://source.datanerd.us/infrastructure/infra-metrics-utils/pull/79) 
- Update `dirac` [PR](https://source.datanerd.us/dirac/dirac/pull/7844)

### Description of `CronJob` kube-state-metrics

The [KSM CronJob metric docs](https://github.com/kubernetes/kube-state-metrics/blob/main/docs/cronjob-metrics.md) do not list descriptions for each metric, but these can be pulled from the KSM `/metrics` endpoint.

Descriptions of newly added metrics:
- `kube_cronjob_labels` Kubernetes labels converted to Prometheus labels.
- `kube_cronjob_info` Info about cronjob.
- `kube_cronjob_created` Unix creation timestamp
- `kube_cronjob_status_active` Active holds pointers to currently running jobs.
- `kube_cronjob_status_last_schedule_time` LastScheduleTime keeps information of when was the last time the job was successfully scheduled.
- `kube_cronjob_status_last_successful_time` LastSuccessfulTime keeps information of when was the last time the job was completed successfully.
- `kube_cronjob_spec_suspend` Suspend flag tells the controller to suspend subsequent executions.
- `kube_cronjob_spec_starting_deadline_seconds` Deadline in seconds for starting the job if it misses scheduled time for any reason.
- `kube_cronjob_next_schedule_time` Next time the cronjob should be scheduled. The time after lastScheduleTime, or after the cron job's creation time if it's never been scheduled. Use this to determine if the job is delayed.
- `kube_cronjob_metadata_resource_version` Resource version representing a specific version of the cronjob.
- `kube_cronjob_spec_successful_job_history_limit` Successful job history limit tells the controller how many completed jobs should be preserved.
- `kube_cronjob_spec_failed_job_history_limit` Failed job history limit tells the controller how many failed jobs should be preserved.

### Testing Plan

✅ Run the local testing environment and check for cronjob metrics in NR1 using the following query: `SELECT * FROM K8sCronjobSample WHERE clusterName = 'htroisi-minikube-k8s-124'`:

```
[
  {
    "results": [
      {
        "events": [
          {
            "clusterName": "htroisi-minikube-k8s-124",
            "concurrencyPolicy": "Allow",
            "createdAt": 1674696653,
            "cronjobName": "e2e-resources-cronjob",
            "displayName": "e2e-resources-cronjob",
            "entityGuid": "MTA4ODYyNDJ8SU5GUkF8TkF8LTUyNDIwNjM3MzgxMjQyMzU5NTk",
            "entityId": "-5242063738124235959",
            "entityKey": "k8s:htroisi-minikube-k8s-124:default:cronjob:e2e-resources-cronjob",
            "entityName": "k8s:htroisi-minikube-k8s-124:default:cronjob:e2e-resources-cronjob",
            "event_type": "K8sCronjobSample",
            "externalKey": "k8s:htroisi-minikube-k8s-124:default:cronjob:e2e-resources-cronjob",
            "integrationName": "com.newrelic.kubernetes",
            "integrationVersion": "dev",
            "isActive": 0,
            "isSuspended": 0,
            "label.cronjob": "e2e-resources-cronjob",
            "label.namespace": "default",
            "lastScheduledTime": 1674696720,
            "metadataResourceVersion": 734,
            "namespaceName": "default",
            "nextScheduledTime": 1674696780,
            "reportingAgent": "nr-nrk8s-ksm-6599866494-f68xn",
            "schedule": "* * * * *",
            "timestamp": 1674696739000
          },
...
```

✅   This PR passes static testing.

✅   This PR passes E2E testing. Note that e2e testing won't pass until the backed PRs (listed above) are merged. This is because e2e testing uses a keyset() query and so until the metrics are shimmed from Sample metrics to Dimensional metrics, the keyset won't find them.

### CODE REVIEW

**Comment by vuqtran88** [comment]
I know updating these static tests can be tedious, but PR with failing or no tests are examples of test smells. Everything that is merged into main branch should be thoroughly tested and ready-to-deploy state.

**Review by sigilioso — COMMENTED** [review]
I agree with @vuqtran88 [comment](https://github.com/newrelic/nri-kubernetes/pull/609#issuecomment-1411149276). I would not merge it until it is properly tested and it is considered ready to be released.

However, it is also true that updating the static test data is time comsuming and working with small PRs is more convenient (because changes are easier to understand and potential problems are more likely to arise). 

I have a suggestion, maybe it is possible to have the best of both worlds if you create a new branch to gather all the changes related to the new k8s objects and you merge this branch there instead of merging it to main. Then, that branch would not be merged into the main branch until all changes are ready and properly tested. WDYT?

**Review by nserrino — APPROVED** [review]

**Comment by nserrino** [comment]
Approving from the perspective of the naming of the metrics. Let's put this as DO NOT MERGE if it's not ready to merge yet

File: e2e/1_16-exceptions.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# This metrics are not present from 1.16
+except_metrics:
+# Added from 1.23
+- k8s.apiserver.currentInflightRequestsReadOnly
```
Comment (htroisi) [inline e2e/1_16-exceptions.yml:4]: Previously, the `e2e.yaml` workflow used the `1_22` exceptions file for testing K8s `1.16`. All of these metric exceptions are from the `1_22` file.

**Review by nserrino — APPROVED** [review]

**Pull request merged**